src/main/java/catbot/task/TaskList.java

```diff
@@ -0,0 +1,108 @@
+package catbot.task;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.function.Consumer;
+
+public class TaskList {
+
+    private ArrayList<Task> tasks;
+    private final String path;
+
+    public TaskList(String path) {
+        this.path = path;
+        this.tasks = new ArrayList<>();
+        readSerializedFromFile();
+    }
+
+
+    public void addTask(Task task) {
+        tasks.add(task);
+        update();
+    }
+
+    public Task removeTask(int index) {
+        Task removed = tasks.remove(index);
+        update();
+        return removed;
+    }
+
+    public void ifValidIndexElse(int index, Consumer<Integer> ifValid, Consumer<Integer> otherwise) {
```

Comment:
Perhaps adding context to the variable names might communicate their purpose better?

Reply:
This is heavily inspired by `Optional`'s [`ifPresentOrElse`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Optional.html#ifPresentOrElse(java.util.function.Consumer,java.lang.Runnable)) (and other similar methods), and I'm trying to take it to its logical extensions; if, even with context, this does not seem intuitive enough, I'm happy to discuss about how to make it better!